sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java

```diff
@@ -39,12 +40,44 @@
 import org.apache.beam.sdk.values.TupleTag;
 
 /**
- * A {@link PTransform} representing an execution plan for a SQL query.
+ * {@link SqlTransform} is the DSL interface of Beam SQL. It translates a SQL query as a {@link
+ * PTransform}, so developers can use standard SQL queries in a Beam pipeline.
  *
- * <p>The table names in the input {@code PCollectionTuple} are only valid during the current query.
+ * <h1>Beam SQL DSL usage:</h1>
+ *
+ * <p>A typical pipeline with Beam SQL DSL is:
+ *
+ * <pre>{@code
+ * PipelineOptions options = PipelineOptionsFactory.create();
+ * Pipeline p = Pipeline.create(options);
+ *
+ * //create table from TextIO;
+ * PCollection<Row> inputTableA = p.apply(TextIO.read().from("/my/input/patha")).apply(...);
+ * PCollection<Row> inputTableB = p.apply(TextIO.read().from("/my/input/pathb")).apply(...);
+ *
+ * //run a simple query, and register the output as a table in BeamSql;
+ * String sql1 = "select MY_FUNC(c1), c2 from PCOLLECTION";
+ * PCollection<Row> outputTableA = inputTableA.apply(
+ *    SqlTransform
+ *        .query(sql1)
+ *        .registerUdf("MY_FUNC", MY_FUNC.class, "FUNC");
+ *
+ * //run a JOIN with one table from TextIO, and one table from another query
+ * PCollection<Row> outputTableB =
+ *     PCollectionTuple
+ *     .of(new TupleTag<>("TABLE_O_A"), outputTableA)
+ *     .and(new TupleTag<>("TABLE_B"), inputTableB)
+ *         .apply(SqlTransform.query("select * from TABLE_O_A JOIN TABLE_B where ..."));
+ *
+ * //output the final result with TextIO
+ * outputTableB.apply(...).apply(TextIO.write().to("/my/output/path"));
+ *
+ * p.run().waitUntilFinish();
+ * }</pre>
  */
 @AutoValue
-public abstract class QueryTransform extends PTransform<PInput, PCollection<Row>> {
+@Experimental
```

Comment:
Just curious, why add @Experimental here?

Reply:
All Beam SQL is experimental at this moment, meaning that APIs can change at any time